File: Data Structures & Algorithms/binary-tree-right-side-view/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public List<Integer> rightSideView(TreeNode root) {

        List<Integer> result = new ArrayList<>();
        if(root == null){
            return result;
        }

        Queue<TreeNode> queue = new ArrayDeque<TreeNode>();
        queue.add(root);

        while(!queue.isEmpty()){
            int levelSize = queue.size();
            for (int i = 0; i<levelSize; i++){
                TreeNode node = queue.poll();
                if(i == 0) result.add(node.val);
                if(node.right != null){
                    queue.add(node.right);
                }

                if(node.left != null){
                    queue.add(node.left);
                }

            }
        }


        return result;

    }
}

File: Data Structures & Algorithms/valid-binary-search-tree/submission-1.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution {
    public bool IsValidBST(TreeNode root) {
        return helper(root, double.NegativeInfinity, double.PositiveInfinity);
        
    }

    private bool helper(TreeNode node, double lower, double upper){
        if(node == null){
            return true;
        }
        if(!((lower < node.val) && node.val < upper)){
            return false;
        }

        return helper(node.left, lower, node.val) && helper(node.right, node.val, upper);
    
    }
}
